src/cdk/table/table.ts

```diff
@@ -828,6 +963,30 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
     this._dataDiffer.diff([]);
     this._rowOutlet.viewContainer.clear();
     this.renderRows();
+    this.updateStickyColumnStyles();
+  }
+
+  /**
+   * Checks if there has been a change in sticky states since last check and applies the correct
+   * sticky styles. Since checking resets the "dirty" state, this should only be performed once
+   * during a change detection and after the inputs are settled (after content check).
+   */
+  private _checkStickyStates() {
+    const stickyCheckReducer = (acc: boolean, d: CdkHeaderRowDef|CdkFooterRowDef|CdkColumnDef) => {
+      return acc || d.checkStickyChanged();
+    };
+
+    if (this._headerRowDefs.reduce(stickyCheckReducer, false)) {
```

Comment:
Is this not just a `some`?
```ts
const hasStickyStateChanged = (d: HasStickyState) => d.checkStickyChanged();

if (this._headerRowDefs.some(hasStickyStateChanged)) {
   ...
}
```

Reply:
No, and I'll add a comment explaining why. The check needs to occur on every definition since it also resets its dirty state